connor/traderModule.go

```diff
@@ -0,0 +1,539 @@
+			return 0, fmt.Errorf("cannot save order to database: %v", err)
+		}
+		log.Printf("Order created ==> ID: %v, Price: %v $, Hashrate: %v H/s \r\n",
+			actOrder.GetId().Unwrap().Int64(),
+			t.PriceToString(actOrder.GetPrice().Unwrap()),
+			actOrder.GetBenchmarks().GPUEthHashrate())
+		return reBuyHash, nil
+	}
+	return buyMgHash, nil
+}
+
+func (t *TraderModule) GetProfitForTokenBySymbol(tokens []*TokenMainData, symbol string) (float64, error) {
+	for _, t := range tokens {
+		if t.Symbol == symbol {
+			return t.ProfitPerMonthUsd, nil
+		}
+	}
+	return 0, fmt.Errorf("Cannot get price from token! ")
+}
+
+func (t *TraderModule) GetPriceForTokenPerSec(token watchers.TokenWatcher, symbol string) (float64, float64, error) {
+	tokens, err := t.profit.CollectTokensMiningProfit(token)
+	if err != nil {
+		return 0, 0, fmt.Errorf("Cannot calculate token prices: %v\r\n", err)
+	}
+	pricePerMonthUSD, err := t.GetProfitForTokenBySymbol(tokens, symbol)
+	if err != nil {
+		return 0, 0, fmt.Errorf("Cannot get profit for tokens: %v\r\n", err)
+	}
+	pricePerSec := pricePerMonthUSD / (secsPerDay * daysPerMonth)
+	return pricePerMonthUSD, pricePerSec, nil
+}
+
+func (t *TraderModule) ReinvoiceOrder(ctx context.Context, cfg *Config, price *sonm.Price, bench map[string]uint64, tag string) error {
+	order, err := t.c.Market.CreateOrder(ctx, &sonm.BidOrder{
+		Duration: &sonm.Duration{Nanoseconds: 0},
+		Price:    price,
+		Tag:      tag,
+		Identity: cfg.OtherParameters.IdentityForBid,
+		Resources: &sonm.BidResources{
+			Network: &sonm.BidNetwork{
+				Overlay:  false,
+				Outbound: true,
+				Incoming: false,
+			},
+			Benchmarks: bench,
+		},
+	})
+	if err != nil {
+		fmt.Printf("Cannot created Lucky Order: %v\r\n", err)
+		return err
+	}
+	if err := t.c.db.SaveOrderIntoDB(&database.OrderDb{
+		OrderID:         order.GetId().Unwrap().Int64(),
+		Price:           order.GetPrice().Unwrap().Int64(),
+		Hashrate:        order.GetBenchmarks().GPUEthHashrate(),
+		StartTime:       time.Now(),
+		ButterflyEffect: int32(OrderStatusReinvoice),
+		ActualStep:      0,
+	}); err != nil {
+		return fmt.Errorf("cannot save reinvoice order %s to DB: %v \r\n", order.GetId().Unwrap().String(), err)
+	}
+	log.Printf("REINVOICE Order ===> %v created (descendant: %v), price: %v, hashrate: %v",
+		order.GetId(), tag, order.GetPrice(), order.GetBenchmarks().GPUEthHashrate())
+	return nil
+}
+
+func (t *TraderModule) CmpChangeOfPrice(change float64, def float64) (int32, error) {
+	if change >= 100+def {
+		return 1, nil
+	} else if change < 100-def {
+		return -1, nil
+	}
+	return 0, nil
+}
+
+func (t *TraderModule) OrdersProfitTracking(ctx context.Context, cfg *Config, actualPrice *big.Int, ordersDb []*database.OrderDb) error {
+	actualPriceCur := big.NewInt(actualPrice.Int64())
+
+	for _, orderDb := range ordersDb {
+		actualPriceClone := big.NewInt(0).Set(actualPriceCur)
+		order, err := t.c.Market.GetOrderByID(ctx, &sonm.ID{Id: strconv.Itoa(int(orderDb.OrderID))})
```

Comment:
fmt.sprintf with %d formatter will be nice here

Reply:
Why?

Reply:
because{call: alot(of(methods)).inOne(line{isNotAwesome: atAll})}